app/components/App/App.js

```diff
@@ -0,0 +1,207 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { autobind } from 'core-decorators';
+import { BrowserRouter, Route, Switch } from 'react-router-dom';
+import findPath from '../../utils/BFS';
+import Game from '../Game/Game';
+// const Settings = require('./Settings');
+// const Results = require('./Results');
+
+function resetBoard(lengthX, lengthY) {
+  console.log('creating board');
+
+  const board = [];
+  for (let i = 0; i < lengthY; i += 1) {
+    board[i] = [];
+    for (let j = 0; j < lengthX; j += 1) {
+      board[i][j] = {
+        isAllowedToBeSteppedOn: true,
+        numberOfStepInPath: -1,
+      };
+    }
+  }
+
+  return board;
+}
+
+function cloneBoard(board) {
+  return (board.map(row => (row.map(cell => ({ ...cell, })))));
+}
+
+@autobind
+export default class App extends React.Component {
+  static get defaultProps() {
+    return {
+      dimensions: {
+        x: 12,
+        y: 12,
+      },
+      colors: {
+        A: 'red',
+        B: 'blue',
+        C: 'green',
+      },
+    };
+  }
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      colorA: props.colors.A,
+      colorB: props.colors.B,
+      colorC: props.colors.C,
+      board: resetBoard(props.dimensions.x, props.dimensions.y),
+      pathStartingPoint: [],
+      pathEndingPoint: [],
+      startButtonPressed: false,
+      endButtonPressed: false,
+      // pathExistsFlag: false, // 'true if a path was found'
+      // time: '',
+    };
+  }
+
+  setBoardColor(updatedColor, newColor) {
+    this.setState(() => {
+      const color = {};
+      color[updatedColor] = newColor;
+      return color;
+    });
+  }
+
+  setStartLocation(x, y) {
+    this.setState({
+      pathStartingPoint: [ x, y ],
+      startButtonPressed: false,
+    });
+  }
+
+  setEndLocation(x, y) {
+    this.setState(() => ({
```

Comment:
No need for function here

Reply:
funnily, no function in `setStartLocation()`.

done